src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -1563,6 +1568,7 @@ public int GetHashCode(ReadOnlySpan<char> source, CompareOptions options)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Is the AggressiveInlining required on all these methods? Many of these are very simple method and the JIT is going to inline them without any hints.

Reply:
I haven't manually checked all of these (do you think it's necessary?), but was basing this on the shape of the `IndexOf` implementation.

There, you have the public `IndexOf` method that calls into [`IndexOfCore`](https://github.com/dotnet/runtime/blob/caa11777b7936df41cc5bdc95f063105f15a452d/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs#L1101), which calls into [`IcuIndexOfCore`](https://github.com/dotnet/runtime/blob/caa11777b7936df41cc5bdc95f063105f15a452d/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs#L63), which finally calls `IndexOfOrdinal{IgnoreCase}Helper`.
Previously, none of these calls were being inlined.
After the change, you end up with the public `IndexOf` method that calls into helpers like `IndexOfOrdinal{IgnoreCase}Helper`.

Reply:
I have done a quick check for this particular one on .NET 7 (without your changes):
```
using System.Globalization;

CompareInfo compareInfo = CultureInfo.CurrentCulture.CompareInfo;

for (;;)
{
    compareInfo.GetHashCode("Hello world", default);
}
```
`set DOTNET_JitDisasm=*Main*`
```
G_M000_IG01:                ;; offset=0000H
       4883EC38             sub      rsp, 56
       4889B424A8000000     mov      qword ptr [rsp+A8H], rsi
       33C0                 xor      eax, eax
       4889442420           mov      qword ptr [rsp+20H], rax
       4889442428           mov      qword ptr [rsp+28H], rax
       488B742470           mov      rsi, gword ptr [rsp+70H]

G_M000_IG02:                ;; offset=001DH
       403836               cmp      byte  ptr [rsi], sil
       48B9684F80FC27010000 mov      rcx, 0x127FC804F68
       488B09               mov      rcx, gword ptr [rcx]
       4883C10C             add      rcx, 12
       488BD6               mov      rdx, rsi
       48894C2420           mov      bword ptr [rsp+20H], rcx
       C74424280B000000     mov      dword ptr [rsp+28H], 11
       488BCA               mov      rcx, rdx
       488D542420           lea      rdx, [rsp+20H]
       4533C0               xor      r8d, r8d
       FF1596781D00         call     [System.Globalization.CompareInfo:IcuGetHashCodeOfString(System.ReadOnlySpan`1[ushort],int):int:this]
       EBC9                 jmp      SHORT G_M000_IG02
```
Both the public `GetHashCode` entrypoint and the `GetHashCodeOfStringCore` are getting inlined without any additional hints.